example/declarative.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2015 Google. All rights reserved. Use of this source code is
+// governed by a BSD-style license that can be found in the LICENSE file.
+
+library grinder.example.declarative;
+
+import 'package:grinder/grinder.dart';
+
+const main = grind;
+
+@Task()
+init(GrinderContext context) {
+  context.log("I set things up");
+}
+
+@Task(depends: const ['init'])
+compile(GrinderContext context) {
+  context.log("I'm compiling now...");
+  //context.fail('woot');
+}
+
+@Task(depends: const ['compile'])
+deploy(GrinderContext context) {
+  context.log("deploying a");
+  context.log("deploying b");
+}
+
+@Task(name: 'default', depends: const ['deploy'])
```

Comment:
Since there can only be one default task, maybe it should be a `Grinder.defaultTask` property, and then we can have an `@DefaultTask()` annotation, and then either add an `isDefault: false` parameter to `addTask`, or a top-level `defaultTask` setter for programmatic use cases.  If you try to set the default task twice, it should fail fast.

You'll still need a separate task declaration when you want your default task to run multiple other tasks.

```dart
@DefaultTask(depends: const [deploy, docs])
all() {}
```